internals/overlord/overlord.go

```diff
@@ -130,12 +159,28 @@ func New(pebbleDir string, restartHandler restart.Handler, serviceOutput io.Writ
 	// Tell service manager about check failures.
 	o.checkMgr.NotifyCheckFailed(o.serviceMgr.CheckFailed)
 
-	// the shared task runner should be added last!
+	if extension != nil {
+		extraManagers, err := o.extension.ExtraManagers(o)
+		if err != nil {
+			return nil, err
+		}
+		for _, manager := range extraManagers {
+			o.addManager(manager)
```

Comment:
nit: I would replace this with ```o.stateEng.AddManager(manager)```. The original purpose of ```addManager``` was to assign manager instances to the overlord struct (snapd), which has been removed. This wrapper adds no value. I think we can clean up (delete the method) the code as part of this?

Reply:
I wouldn't delete the `Overlord.addManager` method in this PR, but I will use `stateEng.AddManager` if you think it's better.